### PR TITLE
Use nvme cli in container instead

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#6798

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM. Waiting for Derek confirmation before merging.

**Comment by derekbit** [comment]
> LGTM. Waiting for Derek confirmation before merging.

Ready for review. Thank you. cc @innobead

File: pkg/nvme/nvmecli.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ type Path struct {
 	State     string `json:"State,omitempty"`
 }
 
-func getNvmeVersion(executor util.Executor) (major, minor int, err error) {
+func NvmeVersion(executor util.Executor) (major, minor int, err error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:63]: It seems this function is not exported, so let's just lower case it instead. `cliVersion`.

File: pkg/nvme/nvmecli.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func getNvmeVersion(executor util.Executor) (major, minor int, err error) {
 	return major, minor, nil
 }
 
-func performNvmeShowHostNQN(executor util.Executor) (string, error) {
+func NvmeShowHostNQN(executor util.Executor) (string, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:101]: `showHostNQN` instead? It seems we are in nvme package already, so the prefix `Nvme` could be optional.

File: pkg/nvme/nvmecli.go [hunk 3]
```diff
@@ -117,8 +117,8 @@ func performNvmeShowHostNQN(executor util.Executor) (string, error) {
 	return "", fmt.Errorf("failed to get host NQN from %s", outputStr)
 }
 
-func performNvmeListSubsystems(device string, executor util.Executor) ([]Subsystem, error) {
-	major, _, err := getNvmeVersion(executor)
+func NvmeListSubsystems(devicePath string, executor util.Executor) ([]Subsystem, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:120]: `listSubsystems`

File: pkg/nvme/nvmecli.go [hunk 4]
```diff
@@ -142,12 +142,12 @@ func performNvmeListSubsystems(device string, executor util.Executor) ([]Subsyst
...
 	if major == 1 {
-		return performNvmeListSubsystemsV1(jsonStr, executor)
+		return NvmeListSubsystemsV1(jsonStr, executor)
 	}
-	return performNvmeListSubsystemsV2(jsonStr, executor)
+	return NvmeListSubsystemsV2(jsonStr, executor)
 }
 
-func performNvmeListSubsystemsV1(jsonStr string, executor util.Executor) ([]Subsystem, error) {
+func NvmeListSubsystemsV1(jsonStr string, executor util.Executor) ([]Subsystem, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:150]: `listSubsystemsV1`

File: pkg/nvme/nvmecli.go [hunk 5]
```diff
@@ -162,17 +162,18 @@ type ListSubsystemsV2Output struct {
 	Subsystems []Subsystem `json:"Subsystems"`
 }
 
-func performNvmeListSubsystemsV2(jsonStr string, executor util.Executor) ([]Subsystem, error) {
+func NvmeListSubsystemsV2(jsonStr string, executor util.Executor) ([]Subsystem, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:165]: `listSubsystemsV2`

File: pkg/nvme/nvmecli.go [hunk 6]
```diff
@@ -189,7 +190,7 @@ type NvmeDevice struct {
 	SectorSize   int32  `json:"SectorSize,omitempty"`
 }
 
-func performNvmeList(executor util.Executor) ([]NvmeDevice, error) {
+func NvmeList(executor util.Executor) ([]NvmeDevice, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:193]: `listDevices`

File: pkg/nvme/nvmecli.go [hunk 7]
```diff
@@ -210,8 +211,8 @@ func performNvmeList(executor util.Executor) ([]NvmeDevice, error) {
 	return output["Devices"], nil
 }
 
-func performNvmeDiscovery(ip, port string, executor util.Executor) ([]DiscoveryPageEntry, error) {
-	hostNQN, err := performNvmeShowHostNQN(executor)
+func NvmeDiscovery(ip, port string, executor util.Executor) ([]DiscoveryPageEntry, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:214]: `discover`

File: pkg/nvme/nvmecli.go [hunk 8]
```diff
@@ -279,8 +280,8 @@ func performNvmeDiscovery(ip, port string, executor util.Executor) ([]DiscoveryP
 	return output.Entries, nil
 }
 
-func performNvmeConnect(ip, port, nqn string, executor util.Executor) (string, error) {
-	hostNQN, err := performNvmeShowHostNQN(executor)
+func NvmeConnect(ip, port, nqn string, executor util.Executor) (string, error) {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:283]: `connect` or `connectController`?

File: pkg/nvme/nvmecli.go [hunk 9]
```diff
@@ -317,7 +318,7 @@ func performNvmeConnect(ip, port, nqn string, executor util.Executor) (string, e
 	return output["device"], nil
 }
 
-func performNvmeDisconnect(nqn string, executor util.Executor) error {
+func NvmeDisconnect(nqn string, executor util.Executor) error {
```
Comment (innobead) [inline pkg/nvme/nvmecli.go:321]: `disconnect`

**Review by innobead — CHANGES_REQUESTED** [review]
Just a few feedback for function naming and conventions.

File: pkg/nvme/nvmecli.go (hunk 1 above)
Comment (derekbit) [reply pkg/nvme/nvmecli.go]: All are renamed. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**